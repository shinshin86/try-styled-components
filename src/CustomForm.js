import React from 'react';
import styled from 'styled-components';
import logo from './logo.svg';

const CustomInput = styled.input.attrs({
  type: props => props.type || 'text',
  margin: props => props.size || '1em',
  padding: props => props.size || '1em'
})`
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;

  type: ${props => props.type};
  margin: ${props => props.margin}
  padding: ${props => props.padding}
`;

const CustomField = styled.p`
  margin: 10px;
  padding: 10px;
  background-color: grey;
`;

const Title = styled.h2`
  border-bottom: solid 1px palevioletred;
`;

const Content = styled.div`
  border: solid 1px grey;
  margin: 10px;
  padding: 10px;
`;

const CustomForm = () => (
  <Content>
    <Title>Custom Form</Title>
    <CustomInput placeholder="text is here" type="text" />
    <CustomInput placeholder="date is here" type="date" />
    <CustomInput placeholder="password is here" type="password" />
    <p>
      <CustomInput placeholder="password is here" type="email" size="3em" />
    </p>
    <CustomField>
      <CustomInput
        placeholder="password is here"
        type="image"
        src={logo}
        alt="react logo"
        width="200"
        height="200"
        size="12px"
      />
    </CustomField>
    <CustomInput type="submit" value="submit" />
  </Content>
);

export default CustomForm;
