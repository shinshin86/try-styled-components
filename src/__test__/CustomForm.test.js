import React from 'react';
import renderer from 'react-test-renderer';
import CustomForm from '../CustomForm';
import 'jest-styled-components';

test('CustomForm Snapshot', () => {
  const tree = renderer.create(<CustomForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('CustomForm have margin and padding 16px', () => {
  const tree = renderer.create(<CustomForm />).toJSON();
  expect(tree).toHaveStyleRule('margin', '16px');
  expect(tree).toHaveStyleRule('padding', '16px');
});
