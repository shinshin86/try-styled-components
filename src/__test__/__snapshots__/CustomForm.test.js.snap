// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomForm Snapshot 1`] = `
.c2 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: text;
  margin: 1em;
  padding: 1em;
}

.c3 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: date;
  margin: 1em;
  padding: 1em;
}

.c4 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: password;
  margin: 1em;
  padding: 1em;
}

.c6 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: image;
  margin: 12px;
  padding: 12px;
}

.c7 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: submit;
  margin: 1em;
  padding: 1em;
}

.c5 {
  margin: 16px;
  padding: 16px;
  background-color: grey;
}

.c1 {
  border-bottom: solid 1px palevioletred;
}

.c0 {
  border: solid 1px grey;
  margin: 16px;
  padding: 16px;
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Custom Form
  </h2>
  <input
    className="c2"
    margin="1em"
    padding="1em"
    placeholder="text is here"
    type="text"
  />
  <input
    className="c3"
    margin="1em"
    padding="1em"
    placeholder="date is here"
    type="date"
  />
  <input
    className="c4"
    margin="1em"
    padding="1em"
    placeholder="password is here"
    type="password"
  />
  <p
    className="c5"
  >
    <input
      alt="react logo"
      className="c6"
      height="200"
      margin="12px"
      padding="12px"
      placeholder="password is here"
      size="12px"
      src="logo.svg"
      type="image"
      width="200"
    />
  </p>
  <input
    className="c7"
    margin="1em"
    padding="1em"
    type="submit"
    value="submit"
  />
</div>
`;
